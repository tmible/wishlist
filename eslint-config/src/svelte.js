import svelte from "eslint-plugin-svelte";
import globals from "globals";

export default [
  ...svelte.configs["flat/recommended"],
  {
    languageOptions: {
      globals: {
        ...globals.browser,
        ...globals.node,
      },
      ecmaVersion: "latest",
      sourceType: "module",
      parserOptions: {
        extraFileExtensions: [ ".svelte" ],
      },
    },

    settings: {
      "import/parsers": {
        "svelte-eslint-parser": [ ".svelte" ],
        "espree": [ ".js" ],
      },
    },

    rules: {
      /* import */
      // svelte does tree shaking when compiling
      "import/no-namespace": "off",

      /* svelte */
      "svelte/infinite-reactive-loop": "error",
      "svelte/no-dom-manipulating": "error",
      "svelte/no-dupe-on-directives": "error",
      "svelte/no-dupe-use-directives": "error",
      "svelte/no-reactive-reassign": "error",
      "svelte/no-store-async": "error",
      "svelte/require-store-callbacks-use-set-param": "error",
      "svelte/block-lang": "error",
      "svelte/no-ignored-unsubscribe": "error",
      "svelte/no-immutable-reactive-statements": "error",
      "svelte/no-inline-styles": "error",
      "svelte/no-reactive-functions": "error",
      "svelte/no-reactive-literals": "error",
      "svelte/no-useless-mustaches": "error",
      "svelte/require-each-key": "error",
      "svelte/require-event-dispatcher-types": "error",
      "svelte/require-stores-init": "error",
      "svelte/valid-each-key": "error",
      "svelte/derived-has-same-inputs-outputs": "error",
      "svelte/first-attribute-linebreak": "error",
      "svelte/html-closing-bracket-new-line": "error",
      "svelte/html-closing-bracket-spacing": "error",
      "svelte/html-quotes": "error",
      "svelte/html-self-closing": [
        "error",
        {
          "void": "never",
          "normal": "never",
          "svg": "always",
          "math": "never",
          "component": "always",
          "svelte": "always",
        },
      ],
      "svelte/indent": [ "error", { ignoredNodes: [ "ConditionalExpression" ] } ],
      "svelte/max-attributes-per-line": [ "error", { multiline: 1, singleline: 100 } ],
      "svelte/mustache-spacing": "error",
      "svelte/no-extra-reactive-curlies": "error",
      "svelte/no-spaces-around-equal-signs-in-attribute": "error",
      "svelte/prefer-class-directive": "error",
      "svelte/prefer-style-directive": "error",
      "svelte/shorthand-attribute": "error",
      "svelte/sort-attributes": [
        "error",
        {
          order: [
            // `this` property.
            "this",
            // `bind:this` directive.
            "bind:this",
            // `id` attribute.
            "id",
            // `name` attribute.
            "name",
            // `slot` attribute.
            "slot",
            // `--style-props`
            { match: "/^--/u", sort: "ignore" },
            // `style` attribute, and `style:` directives.
            [ "style", "/^style:/u" ],
            // `class` attribute.
            "class",
            // `class:` directives.
            { match: "/^class:/u", sort: "ignore" },
            // other attributes.
            {
              match: [ "!/:/u", "!/^(?:this|id|name|style|class)$/u", "!/^--/u" ],
              sort: "ignore",
            },
            // `bind:` directives (other then `bind:this`), and `on:` directives.
            [ "/^bind:/u", "!bind:this", "/^on:/u" ],
            // `use:` directives.
            { match: "/^use:/u", sort: "ignore" },
            // `transition:` directive.
            { match: "/^transition:/u", sort: "ignore" },
            // `in:` directive.
            { match: "/^in:/u", sort: "ignore" },
            // `out:` directive.
            { match: "/^out:/u", sort: "ignore" },
            // `animate:` directive.
            { match: "/^animate:/u", sort: "ignore" },
            // `let:` directives.
            { match: "/^let:/u", sort: "ignore" },
          ],
        },
      ],
      "svelte/spaced-html-comment": "error",
      "svelte/no-inner-declarations": "error",
      "svelte/no-trailing-spaces": "error",
    },
  },
  {
    files: [ "**/*.svelte" ],
    rules: {
      // svelte components can import themselves
      "import/no-self-import": "off",
    },
  },
  {
    files: [ "**/**.test.js", "**/**.test.svelte.js", "**/__tests__/**" ],
    settings: { "import/ignore": [ "vitest", "@testing-library/svelte" ] },
  },
];
