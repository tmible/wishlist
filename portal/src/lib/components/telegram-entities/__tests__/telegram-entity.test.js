// @vitest-environment jsdom
import { cleanup, render } from '@testing-library/svelte';
import { afterEach, beforeEach, describe, expect, it } from 'vitest';
import TelegramEntity from '../telegram-entity.svelte';

describe('telegram entity', () => {
  let node;

  beforeEach(() => {
    node = {
      type: 'type',
      children: [],
    };
  });

  afterEach(cleanup);

  it('should display bold', () => {
    node.type = 'bold';
    const { container } = render(TelegramEntity, { node });
    expect(container.innerHTML).toMatchSnapshot();
  });

  it('should display italic', () => {
    node.type = 'italic';
    const { container } = render(TelegramEntity, { node });
    expect(container.innerHTML).toMatchSnapshot();
  });

  it('should display underline', () => {
    node.type = 'underline';
    const { container } = render(TelegramEntity, { node });
    expect(container.innerHTML).toMatchSnapshot();
  });

  it('should display strikethrough', () => {
    node.type = 'strikethrough';
    const { container } = render(TelegramEntity, { node });
    expect(container.innerHTML).toMatchSnapshot();
  });

  it('should display code', () => {
    node.type = 'code';
    const { container } = render(TelegramEntity, { node });
    expect(container.innerHTML).toMatchSnapshot();
  });

  it('should display blockquote', () => {
    node.type = 'blockquote';
    const { container } = render(TelegramEntity, { node });
    expect(container.innerHTML).toMatchSnapshot();
  });

  describe('pre', () => {
    beforeEach(() => node.type = 'pre');

    it('should display pre without language', () => {
      const { container } = render(TelegramEntity, { node });
      expect(container.innerHTML).toMatchSnapshot();
    });
    it('should display pre with language', () => {
      node.language = 'javascript';
      const { container } = render(TelegramEntity, { node });
      expect(container.innerHTML).toMatchSnapshot();
    });
  });

  describe('url', () => {
    beforeEach(() => node.type = 'url');

    it('should display url without protocol', () => {
      node.children = [{ text: 'example.com' }];
      const { container } = render(TelegramEntity, { node });
      expect(container.innerHTML).toMatchSnapshot();
    });

    it('should display url with http protocol', () => {
      node.children = [{ text: 'http://example.com' }];
      const { container } = render(TelegramEntity, { node });
      expect(container.innerHTML).toMatchSnapshot();
    });

    it('should display url with https protocol', () => {
      node.children = [{ text: 'https://example.com' }];
      const { container } = render(TelegramEntity, { node });
      expect(container.innerHTML).toMatchSnapshot();
    });
  });

  describe('text_link', () => {
    beforeEach(() => node.type = 'text_link');

    it('should display text_link without url', () => {
      const { container } = render(TelegramEntity, { node });
      expect(container.innerHTML).toMatchSnapshot();
    });

    it('should display text_link with url', () => {
      node.url = 'https://example.com';
      const { container } = render(TelegramEntity, { node });
      expect(container.innerHTML).toMatchSnapshot();
    });
  });

  it('should display spoiler', () => {
    node.type = 'spoiler';
    const { container } = render(TelegramEntity, { node });
    expect(container.innerHTML).toMatchSnapshot();
  });

  it('should display mention', () => {
    node.type = 'mention';
    node.children = [{ text: '@mention' }];
    const { container } = render(TelegramEntity, { node });
    expect(container.innerHTML).toMatchSnapshot();
  });

  it('should display hashtag', () => {
    node.type = 'hashtag';
    const { container } = render(TelegramEntity, { node });
    expect(container.innerHTML).toMatchSnapshot();
  });

  it('should display bot_command', () => {
    node.type = 'bot_command';
    const { container } = render(TelegramEntity, { node });
    expect(container.innerHTML).toMatchSnapshot();
  });

  it('should display cashtag', () => {
    node.type = 'cashtag';
    const { container } = render(TelegramEntity, { node });
    expect(container.innerHTML).toMatchSnapshot();
  });

  it('should display email', () => {
    node.type = 'email';
    node.children = [{ text: 'email' }];
    const { container } = render(TelegramEntity, { node });
    expect(container.innerHTML).toMatchSnapshot();
  });

  it('should display phone_number', () => {
    node.type = 'phone_number';
    node.children = [{ text: 'phone_number' }];
    const { container } = render(TelegramEntity, { node });
    expect(container.innerHTML).toMatchSnapshot();
  });

  it('should display paragraph', () => {
    node.type = 'paragraph';
    const { container } = render(TelegramEntity, { node });
    expect(container.innerHTML).toMatchSnapshot();
  });

  it('should display mention_name', () => {
    node.type = 'mention_name';
    node.text = '  text\n ';
    const { container } = render(TelegramEntity, { node });
    expect(container.innerHTML).toMatchSnapshot();
  });

  it('should display bank_card', () => {
    node.type = 'bank_card';
    node.text = '  text\n ';
    const { container } = render(TelegramEntity, { node });
    expect(container.innerHTML).toMatchSnapshot();
  });

  it('should display custom_emoji', () => {
    node.type = 'custom_emoji';
    node.text = '  text\n ';
    const { container } = render(TelegramEntity, { node });
    expect(container.innerHTML).toMatchSnapshot();
  });

  it('should display default', () => {
    node.text = 'text';
    const { container } = render(TelegramEntity, { node });
    expect(container.innerHTML).toMatchSnapshot();
  });

  it('should display children recursively', () => {
    node.type = 'bold';
    node.children = [{
      type: 'italic',
      children: [{
        type: 'underline',
        children: [{ text: 'formatted' }],
      }],
    }];
    const { container } = render(TelegramEntity, { node });
    expect(container.innerHTML).toMatchSnapshot();
  });
});
