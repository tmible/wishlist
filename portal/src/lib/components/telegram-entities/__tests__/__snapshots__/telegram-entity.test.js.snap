// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`telegram entity > pre > should display pre with language 1`] = `"<pre language="javascript"></pre>"`;

exports[`telegram entity > pre > should display pre without language 1`] = `"<pre></pre>"`;

exports[`telegram entity > should display bank_card 1`] = `"text"`;

exports[`telegram entity > should display blockquote 1`] = `"<blockquote class="prose"></blockquote>"`;

exports[`telegram entity > should display bold 1`] = `"<b></b>"`;

exports[`telegram entity > should display bot_command 1`] = `"<span class="link cursor-auto" data-entity="bot_command"></span>"`;

exports[`telegram entity > should display cashtag 1`] = `"<span class="link cursor-auto" data-entity="cashtag"></span>"`;

exports[`telegram entity > should display children recursively 1`] = `"<b><i><u>formatted</u></i></b>"`;

exports[`telegram entity > should display code 1`] = `"<code></code>"`;

exports[`telegram entity > should display custom_emoji 1`] = `"text"`;

exports[`telegram entity > should display default 1`] = `"text"`;

exports[`telegram entity > should display email 1`] = `"<a href="mailto:email" data-entity="email">email</a>"`;

exports[`telegram entity > should display hashtag 1`] = `"<span class="link cursor-auto" data-entity="hashtag"></span>"`;

exports[`telegram entity > should display italic 1`] = `"<i></i>"`;

exports[`telegram entity > should display mention 1`] = `"<a href="https://t.me/mention" data-entity="mention">@mention</a>"`;

exports[`telegram entity > should display mention_name 1`] = `"text"`;

exports[`telegram entity > should display paragraph 1`] = `"<p class="break-words"></p>"`;

exports[`telegram entity > should display phone_number 1`] = `"<a href="tel:phone_number" data-entity="phone_number">phone_number</a>"`;

exports[`telegram entity > should display spoiler 1`] = `"<span class="text-transparent hover:text-base-content spoiler" data-entity="spoiler"></span>"`;

exports[`telegram entity > should display strikethrough 1`] = `"<s></s>"`;

exports[`telegram entity > should display underline 1`] = `"<u></u>"`;

exports[`telegram entity > text_link > should display text_link with url 1`] = `"<a href="https://example.com" data-entity="text_link"></a>"`;

exports[`telegram entity > text_link > should display text_link without url 1`] = `"<a data-entity="text_link"></a>"`;

exports[`telegram entity > url > should display url with http protocol 1`] = `"<a href="https://example.com">http://example.com</a>"`;

exports[`telegram entity > url > should display url with https protocol 1`] = `"<a href="https://example.com">https://example.com</a>"`;

exports[`telegram entity > url > should display url without protocol 1`] = `"<a href="https://example.com">example.com</a>"`;
